control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3927,9 +3930,65 @@ func (r *HostedControlPlaneReconciler) reconcileDefaultSecurityGroup(ctx context
 		return nil
 	}
 	if hcp.Status.Platform != nil && hcp.Status.Platform.AWS != nil && hcp.Status.Platform.AWS.DefaultWorkerSecurityGroupID != "" {
-		// Security group has already been created, nothing to do
+		//Security group has already been created, nothing to do
+		var missingPermissions []*ec2.IpPermission
+		var actualPermissions []*ec2.IpPermission
+
+		expectedPermissions, sgID, err := getDefaultWorkerSGIngressRules(r.ec2Client, ctx, hcp.Spec.InfraID, hcp.Spec.Platform.AWS.CloudProviderConfig.VPC)
+		if err != nil {
+			return err
+		}
+
+		hcpAnnotations := hcp.GetAnnotations()
+
+		if val, ok := hcpAnnotations[IPPermissionsAnnotationKey]; ok && val != "" {
+			err := json.Unmarshal([]byte(hcpAnnotations[IPPermissionsAnnotationKey]), &actualPermissions)
+			if err != nil {
+				return err
+			}
+			for _, expectedPermission := range expectedPermissions {
```

Comment:
What is the source of truth for the ingress rules: the HCP annotation or what comes back from `getDefaultWorkerSGIngressRules`?

I was just wondering if someone updated the rules on the HCP annotation, those would get missed in this check right? This is making sure all the expected permissions are in the list of actual permissions.

Reply:
`getDefaultWorkerSGIngressRules` is the source of truth, so yes if someone added an annotation it would get missed.